### PR TITLE
Allow updating snapshots with the `UPDATE_SNAPSHOTS` environment variable

### PR DESCRIPTION
**Pull request by timrogers**
Jest [allows][1] you to update your snapshots by running with the `-u` or `--updateSnapshot` command line parameters.

This adds similar functionality to `rspec-snapshot`, except using the `UPDATE_SNAPSHOTS` environment variable, since RSpec [doesn't support][2] custom arguments to the `rspec` binary.

In the process, I've also elected to improve the README's explanation of how the gem works, and I've improved the tests, both in terms of structure and coverage (they now actually check that any existing snapshot is used!).

This is a new version of my previous PR, #9, but against the new `next` branch.

[1]: https://facebook.github.io/jest/docs/en/snapshot-testing.html
[2]: https://github.com/rspec/rspec-core/issues/2258

### CODE REVIEW

**Comment by timrogers** [comment]
@yesmeck Ready for review!

File: spec/rspec/snapshot/matchers_spec.rb [hunk 1]
```diff
@@ -3,25 +3,84 @@
...
 
-  context "when snapshotting non-string objects" do
-    it "stringifies simple POROs" do
-      simple_data_structure = { a_key: %w(some values) }
-      expect(simple_data_structure).to match_snapshot("simple_data_structure")
+  context "when the snapshot doesn't exist yet" do
+    context "with a hash" do
+      it "stores the value as a snapshot" do
+        hash = { a: 1, b: 2 }
+        expect(hash).to match_snapshot("snapshot/hash")
```
Comment (yesmeck) [inline spec/rspec/snapshot/matchers_spec.rb]: `expect(hash).to match_snapshot("hash")`

**Comment by timrogers** [comment]
Oops! I've fixed that now.

**Comment by Carlows** [comment]
Would be nice to have this merged. Feels like this should have been there in the first place, otherwise how do you update the snapshot? deleting it manually? 🤔

**Pull request merged**